cmd/gen-go-sample/main.go

```diff
@@ -225,77 +229,68 @@ func (g *generator) genSample(ifaceName string, methConf GAPICMethod, regTag str
 	}
 	g.imports[inSpec] = true
 
-	initInfo, err := g.getInitInfo(inType, methConf, valSet)
-	argNames := initInfo.argNames
-	argTrees := initInfo.argTrees
-	flagNames := initInfo.flagNames
-	files := initInfo.files
-	itree := initInfo.reqTree
-
-	if err := g.sampFuncAndClient(serv.GetName(), meth.GetName(), initInfo, regTag); err != nil {
+	initInfo, err := g.getInitInfo(inType, methConf, sampConf.Request)
+	if err != nil {
 		return err
 	}
 
-	// Set up the request
-	{
-		var buf bytes.Buffer
-
-		for i, name := range argNames {
-			fmt.Fprintf(&buf, "%s := ", name)
-			if err := argTrees[i].Print(&buf, g); err != nil {
-				return errors.E(err, "can't initialize parameter: %s", name)
-			}
-			buf.WriteByte('\n')
-		}
-		prependLines(&buf, "// ", false)
-
-		for _, info := range files {
-			handleReadFile(info, &buf, g)
-		}
+	p := g.pt.Printf
 
-		buf.WriteString("req := ")
-		if err := itree.Print(&buf, g); err != nil {
-			return errors.E(err, "can't initialize request object")
-		}
-		buf.WriteByte('\n')
-		prependLines(&buf, "\t", true)
+	// Region tag, function signature and new client
+	argStr, err := argListStr(initInfo, g)
+	if err != nil {
+		return err
+	}
+	p("// [START %s]", sampConf.RegionTag)
+	p("")
+	p("func sample%s(%s) error {", meth.GetName(), argStr)
+	p("  ctx := context.Background()")
+	p("  c, err := %s.New%sClient(ctx)", g.clientPkg.Name, pbinfo.ReduceServName(serv.GetName(), g.clientPkg.Name))
+	p("  if err != nil {")
+	p("    return err")
+	p("  }")
+	p("")
+	g.imports[pbinfo.ImportSpec{Path: "context"}] = true
 
-		if _, err := buf.WriteTo(g.pt.Writer()); err != nil {
-			return err
-		}
+	// Set up the request object
+	if err := g.handleRequest(initInfo); err != nil {
+		return err
 	}
 
 	// Make the RPC call and handle output
 	if meth.GetOutputType() == ".google.protobuf.Empty" {
-		err = g.emptyOut(meth, valSet)
+		err = g.emptyOut(meth, sampConf.Response)
 	} else if meth.GetOutputType() == ".google.longrunning.Operation" {
-		err = g.lro(meth, methConf, valSet)
+		err = g.lro(meth, methConf, sampConf.Response)
 	} else if meth.GetServerStreaming() || meth.GetClientStreaming() {
 		err = errors.E(nil, "streaming methods not supported yet")
```

Comment:
This is simply due to us not implementing them yet in samplegen? If so, could you add a TODO here (so this is easy to find) referencing a GitHub issue to do this? Thanks!

Reply:
https://github.com/googleapis/gapic-generator-go/issues/177 Will also link in code.